### PR TITLE
WIP: Bio 1506 mock dsmc

### PR DESCRIPTION
**Pull request by mariya**
**What problems does this PR solve?**
This mocks the downloading of an archive from PDC, allowing the bypassing of dsmc commands. This in turn allows the md5 validation of a locally available archive in environments where dsmc is not available, e.g. a development environment.

**How has the changes been tested?**
Running automated tests; manual testing by downloading an archive from staging and running the verification service on it locally. Additional unit tests have been written to ensure the correct PdcClient is instantiated based on a config variable.

NOTE: since I do not have rights to deploy to staging, I have been unable to conduct any manual regression testing for actual dsmc downloads. I need the help of someone with sysadmin rights to deploy to staging so manual testing can be done there.

**Reasons for careful code review**
If any of the boxes below are checked, extra careful code review should be inititated.

  - [ ] This PR contains code that could remove data

### CODE REVIEW

**Comment by mariya** [comment]
Changing this to a WIP, this branch should be tested on staging before merge. @johandahlberg maybe you can help me deploy it if this otherwise looks ok?

File: archive_verify/pdc_client.py
Comment (johandahlberg) [inline archive_verify/pdc_client.py:40]: Maybe we should raise the log levels here to info? Feels like it could be interesting to see even when not debugging?

File: archive_verify/pdc_client.py
Comment (johandahlberg) [inline archive_verify/pdc_client.py]: Same here, maybe info rather than debug warning levels?

File: archive_verify/workers.py
Comment (johandahlberg) [inline archive_verify/workers.py:41]: I'm not sure if this needs to have a default value, but that is probably just a mater of taste.

File: archive_verify/workers.py
Comment (johandahlberg) [inline archive_verify/workers.py]: I would typically call this type of construct `pdc_client_factory` or something similar, but that might just be to much time in Java speaking. I think it makes the intention somewhat clearer.

File: config/app.yaml
Comment (johandahlberg) [inline config/app.yaml]: Should this really default to a directory on root? I think for development purposes it might be better if it defaulted to `data/verify/` as it was before.

File: config/app.yaml
Comment (johandahlberg) [inline config/app.yaml:20]: I'd rather see the real client being the default here.

**Review by johandahlberg — APPROVED** [review]
Great work. Some minor comments that you can decide if you want to do anything about or not, but I think it looks ready to merge if you think it is.

File: archive_verify/pdc_client.py
Comment (mariya) [reply archive_verify/pdc_client.py:40]: Agree, fixing.

File: archive_verify/pdc_client.py
Comment (mariya) [reply archive_verify/pdc_client.py]: Fixing.

File: archive_verify/workers.py
Comment (mariya) [reply archive_verify/workers.py:41]: Probably overkill, but I like keeping it there for clarity.

File: archive_verify/workers.py
Comment (mariya) [reply archive_verify/workers.py]: I was struggling with the naming here, good suggestion!

File: config/app.yaml
Comment (mariya) [reply config/app.yaml:20]: As I understand app.yaml is by default geared to development environments, since staging and prod configs are managed by system-management? If so, shouldn't we check in the development-oriented config here? Or am I misunderstanding the setup?

**Comment by mariya** [comment]
Changing to WIP; let's hold off on merging until we get a chance to do the regression testing on staging.

File: config/app.yaml
Comment (mariya) [reply config/app.yaml]: Fixed.

File: config/app.yaml
Comment (johandahlberg) [reply config/app.yaml:20]: Well, you're right. That does make more sense. 👍

**Comment by mariya** [comment]
@johandahlberg @b97pla can one of you please add a pair of eyeballs to the latest commit, which actually performs the post-verification cleanup? You can view those changes and comments separately in my "internal" PR: https://github.com/mariya/snpseq-archive-verify/pull/1

**Review by b97pla — APPROVED** [review]
Great work! This looks good to me :+1:

**Pull request merged**
